src/adm/filter.rs

```diff
@@ -477,15 +507,47 @@ mod tests {
             .check_advertiser(&settings, &mut tile, &mut tags)
             .is_err());
 
-        //good, extra element in host
+        //Good, extra element in host
         tile.advertiser_url = "https://black_friday.acme.biz/ca/".to_owned();
         assert!(filter
             .check_advertiser(&settings, &mut tile, &mut tags)
             .is_ok());
 
+        //Good, extra matching
+        tile.advertiser_url = "https://acme.biz/usa".to_owned();
+        assert!(filter
+            .check_advertiser(&settings, &mut tile, &mut tags)
+            .is_ok());
+
+        // Bad, path doesn't match exactly
+        tile.advertiser_url = "https://acme.biz/usa/".to_owned();
+        assert!(filter
+            .check_advertiser(&settings, &mut tile, &mut tags)
+            .is_err());
+
         // "Traditional host. "
         let s = r#"{
-            "advertiser_hosts": ["acme.biz/ca", "www.acme.co/foo.bar"],
+            "advertiser_urls": [
+                {
+                    "host": "acme.biz",
+                    "paths": [
+                        { "value": "/ca", "matching": "prefix" }
```

Comment:
shouldn't these panic? https://github.com/mozilla-services/contile/pull/311/files#diff-34387244eabf61c959f3a7e535ca04a551c779b9ee645d2e93303c0020f5a881R50

Reply:
Those "values" are wrong, they should have trailing slashes.